### ISSUE TITLE
After updating to 5.0.0 all JS files return `JS_SYNTAX_ERROR` on `npx web-ext lint`

### ISSUE THREAD

**Issue by birtles** [mozilla/web-ext#1976]
### Is this a feature request or a bug?

Bug

### What is the current behavior?

I haven't had a chance to produce a minimized STR yet but I'm filing this in case anyone else faces the same problem.

After updating to web-ext 5.0.0 none of my JS files validate. Instead I get errors such as the following:

```log
JS_SYNTAX_ERROR   JavaScript syntax   There is a JavaScript syntax error in your   rikaichamp-options.js
                  error               code; validation cannot continue on this
                                      file.
```

After digging into what is causing the error I see `espree` is throwing a `Invalid ecmaVersion` error.

Further digging in it seems `addons-linter` is `2020` to espree's `normalizeEcmaVersion` ([source](https://github.com/mozilla/addons-linter/blob/c9d172033092ded469f3e83ba57b34c5048e8dd2/src/scanners/javascript.js#L13)).

`espree` 7.2.0 supports this and `addons-linter` appears to correctly require 7.2.0. Furthermore, within `node_modules\addons-linter\node_modules` we are getting version 7.2.0.

However, when I run `npx web-ext lint` (or `yarn web-ext link` etc.) it seems to pick up the version of `espree` in the root `node_modules` folder which is version 3.5.4. That in turns appears to be required by a chain of dependencies that goes back to `addons-linter` itself.

I'm filing this under `web-ext` however since I suspect this only affects people using `web-ext` as a local package as opposed to when it is globally installed.

I'm well out of my depth when it comes to node package management but if I get a chance I'll make a minified STR for this.

Basically, it seems that the updated `addons-linter` in 5.0.0 is passing `ECMASCRIPT

### Version information (for bug reports)

* **Firefox version**: n/a
* **Your OS and version**: Windows 10 (using Git bash) although this also reproduces on Linux
* Paste the output of these commands:

````
node --version && npm --version && web-ext --version
````

```log
v12.16.3
6.14.4
bash: web-ext: command not found
```

(Note that I am using yarn 1.19.1.)

**Comment by rpl** [comment]
Hi @birtles 
I took a look into this by reproducing it using your add-on repo (birtles/rikaichamp + the changes in birtles/rikaichamp#332) and it turned out that the issue is triggered because of the web-ext-webpack-plugin dev dependency:

- web-ext-webpack-plugin directly depends from an older web-ext version
- the older web-ext version depends from an older version of the addons-linter version and this makes two different addons-linter versions to be listed in the yarn.lock file, 1.0.0 and 2.1.0 after birtles/rikaichamp#332, 1.0.0 and 1.26.0 in the current master:
  - https://github.com/birtles/rikaichamp/blob/ab51ad719e51c2734daab74619110d71b9455702/yarn.lock#L1517
  - https://github.com/birtles/rikaichamp/blob/ab51ad719e51c2734daab74619110d71b9455702/yarn.lock#L1560
 (and this then brings into node_modules an older eslint version and the related older espree version)

This is now my favorite "node packages dependencies hell" issue of the month :-)

I quickly tried to just (temporarily) remove web-ext-webpack-plugin from the rikaichamp devDependencies and refresh the node_modules dir using `yarn install` and at that point `web-ext lint` does pass successfully as it should, and so it should be enough to update the `web-ext` dependency in the `web-ext-webpack-plugin` package to fix the issue in your repo.

Based on the above analysis, this issue may be invalid from a web-ext standpoint, but I'll keep this open for now, so that you can give a try and confirm that what I described above is right and you have been able to fix the issue as expected, and also in case this may be a more common scenario than we may expect and other users may run into it.

**Comment by birtles** [comment]
Hi @rpl!

Thank you! You solved it! That would have taken me a lot longer to work out. Thank you again!

I agree this issue is invalid from a web-ext standpoint but I hope filing it proves useful to anyone else who experiences similar issues.
